compiler/src/dotty/tools/dotc/ast/Positioned.scala

```diff
@@ -68,7 +68,9 @@ abstract class Positioned(implicit @constructorOnly src: SourceFile) extends Src
         if (mySpan.isSynthetic) {
           if (!mySpan.exists && span.exists)
             envelope(source, span.startPos) // fill in children spans
-          this
+            this
+          else
+            cloneIn(source)
         }
         else cloneIn(source)
```

Comment:
I think this can be simplified to:
```scala
        if !mySpan.exists then
          if span.exists then envelope(source, span.startPos) // fill in children spans
          this
        else
          cloneIn(source)
```